File: Basics/binarysearch.java
import java.io.*;
import java.util.*;

public class Main{
  public static void Binarysearch(int[] arr, int data) {

    int si = 0; ei = arr.length -1;
    while(si <= ei){
        int mid = (si+ei)/2;
        if(arr [mid] == data){
            return mid;
        } elseif(data < arr[mid]) {
            ei = mid - 1;
        
        } else si = mid +1;
    }
    return - 1;
  }

public static void main(String[] args) {
    

}

File: Time & space/Binary Search.cpp
#include<iostream.h>
#include<vector>

using namespace std;

int BinarySearch(vector<int> &arr, int data, int si, int ei){

    while(si <= ei){
        int mid = (si + ei) / 2;

        if(arr[mid] == data)
            return mid;
        else if(arr[mid] < data)
            si = mid + 1;
        else
        {
            ei = mid - 1;
        }
        
    }

    return - 1;
}

